core/Common.php

```diff
@@ -263,33 +261,9 @@ public static function mb_strtolower($string)
      */
     public static function sanitizeInputValues($value, $alreadyStripslashed = false)
     {
-        if (is_numeric($value)) {
-            return $value;
-        } elseif (is_string($value)) {
-            $value = self::sanitizeString($value);
-
-            if (!$alreadyStripslashed) {
-                // a JSON array was already stripslashed, don't do it again for each value
-
-                $value = self::undoMagicQuotes($value);
-            }
-        } elseif (is_array($value)) {
-            foreach (array_keys($value) as $key) {
-                $newKey = $key;
-                $newKey = self::sanitizeInputValues($newKey, $alreadyStripslashed);
-                if ($key != $newKey) {
-                    $value[$newKey] = $value[$key];
-                    unset($value[$key]);
-                }
-
-                $value[$newKey] = self::sanitizeInputValues($value[$newKey], $alreadyStripslashed);
-            }
-        } elseif (!is_null($value)
-            && !is_bool($value)
-        ) {
-            throw new Exception("The value to escape has not a supported type. Value = " . var_export($value, true));
-        }
-        return $value;
+        /** @var RequestSanitizer $requestSanitizer */
+        $requestSanitizer = StaticContainer::get('Piwik\Http\RequestSanitizer');
+        return $requestSanitizer->sanitizeValues($value, $alreadyStripslashed);
```

Comment:
Tracking 600 requests takes 10.5s for me on master and ~11.5s on this branch. According to xhprof, it's not the `StaticContainer::get` call that's causing a problem, but that `Common::getRequestVar` is called twice as many times. Not sure why... maybe a faulty rebase. Still investigating.